Issue 14984: Remove divide by zero error in index_put_

jotsif opened the pull request:
No check for zero index tensor was done in the accumulate=True (serial) case in the new TensorIterator code since https://github.com/pytorch/pytorch/pull/13420.

https://github.com/pytorch/pytorch/issues/14984

jotsif committed:
Remove divide by zero error

jotsif committed:
Introduce simular check for zero index in serial_for_each as in for_each

colesbury reviewed (commented):
Thanks for fixing this.

You fixed this in two different ways:
https://github.com/pytorch/pytorch/pull/14986/commits/9bc32c6d968e22030bc11ebc1129de5f6c710c56

and

https://github.com/pytorch/pytorch/pull/14986/commits/fe2c31e67d386e7cd4622bb835433f7c14bfd520

The first fix seems more robust to me. Why did you prefer the second method?

Would you please also add a test case?

jotsif committed:
Add test of index_put_

colesbury committed:
Use range.size() instead of range.end

colesbury committed:
Guard division by size

Included from https://github.com/pytorch/pytorch/pull/14986/commits/9bc32c6d968e22030bc11ebc1129de5f6c710c56

colesbury reviewed (approved):
Thanks! I included your fix from https://github.com/pytorch/pytorch/pull/14986/commits/9bc32c6d968e22030bc11ebc1129de5f6c710c56 as well
